x-pack/plugins/fleet/server/saved_objects/model_versions/security_solution/v15_advanced_package_policy_fields.test.ts

```diff
@@ -0,0 +1,157 @@
+      type: getSavedObjectTypes()[PACKAGE_POLICY_SAVED_OBJECT_TYPE],
+    });
+
+    policyConfigSO = {
+      id: 'mock-saved-object-id',
+      attributes: {
+        name: 'Some Policy Name',
+        package: {
+          name: 'endpoint',
+          title: '',
+          version: '',
+        },
+        id: 'endpoint',
+        policy_id: '',
+        policy_ids: [],
+        enabled: true,
+        namespace: '',
+        revision: 0,
+        updated_at: '',
+        updated_by: '',
+        created_at: '',
+        created_by: '',
+        inputs: [
+          {
+            type: 'endpoint',
+            enabled: true,
+            streams: [],
+            config: {
+              policy: {
+                value: {
+                  windows: {},
+                  mac: {},
+                  linux: {},
+                },
+              },
+            },
+          },
+        ],
+      },
+      type: PACKAGE_POLICY_SAVED_OBJECT_TYPE,
+      references: [],
+    };
+  });
+
+  /** Builds object key paths for all parent objects
+   *
+   * @param path e.g. `advanced.events.optionName`
+   * @returns e.g. ['advanced', 'advanced.events']
+   */
+  const getParentObjectKeyPaths = (path: string): string[] =>
+    path
+      .split('.') // ['advanced', 'events', 'optionName']
+      .slice(0, -1) // ['advanced', 'events']
+      .map((parentObject) => path.match(`^.*${parentObject}`)![0]); // ['advanced', 'advanced.events']
+
+  describe(`when updating to model version ${TARGET_MODEL_VERSION}`, () => {
+    describe.each`
+      name                               | path                                        | backfill
+      ${'aggregate_process'}             | ${'advanced.events.aggregate_process'}      | ${false}
+      ${'set_extended_host_information'} | ${'advanced.set_extended_host_information'} | ${true}
+      ${'alerts.hash.md5'}               | ${'advanced.alerts.hash.md5'}               | ${true}
+      ${'alerts.hash.sha1'}              | ${'advanced.alerts.hash.sha1'}              | ${true}
+      ${'events.hash.md5'}               | ${'advanced.events.hash.md5'}               | ${true}
+      ${'events.hash.sha1'}              | ${'advanced.events.hash.sha1'}              | ${true}
+    `(
+      'backfilling `$name` with `$backfill`',
+      ({ path, backfill }: { path: string; backfill: boolean }) => {
+        it('should backfill when there are no advanced options yet', () => {
+          const migratedPolicyConfigSO = migrator.migrate<PackagePolicy, PackagePolicy>({
+            document: policyConfigSO,
+            fromVersion: TARGET_MODEL_VERSION - 1,
+            toVersion: TARGET_MODEL_VERSION,
+          });
+
+          const migratedPolicyConfig = getConfig(migratedPolicyConfigSO);
+
+          expectConfigToHave(migratedPolicyConfig, path, backfill);
+        });
+
+        it.each(getParentObjectKeyPaths(path))(
+          'should backfill without modifying other options in parent object `%s`',
```

Comment:
the point here to catch all shenanigans when backfilling fields in the prod code using spread operators, so every parent object is checked for additional fields, and this also checks what happens if the objects below don't exist (in other words, if you missed an optional chaining in prod code)

the test cases will look like this
```
  Defend integration advanced policy fields v8.16.0
    when updating to model version 15
      backfilling `aggregate_process` with `false`
        ✓ should backfill when there are no advanced options yet (20 ms)
        ✓ should backfill without modifying other options in parent object `advanced` (3 ms)
        ✓ should backfill without modifying other options in parent object `advanced.events` (2 ms)
        ✓ should not backfill if field is already present (2 ms)
      backfilling `set_extended_host_information` with `true`
        ✓ should backfill when there are no advanced options yet (1 ms)
        ✓ should backfill without modifying other options in parent object `advanced` (2 ms)
        ✓ should not backfill if field is already present (1 ms)
      backfilling `alerts.hash.md5` with `true`
        ✓ should backfill when there are no advanced options yet (3 ms)
        ✓ should backfill without modifying other options in parent object `advanced` (2 ms)
        ✓ should backfill without modifying other options in parent object `advanced.alerts` (2 ms)
        ✓ should backfill without modifying other options in parent object `advanced.alerts.hash` (1 ms)
        ✓ should not backfill if field is already present
      backfilling `alerts.hash.sha1` with `true`
        ✓ should backfill when there are no advanced options yet
        ✓ should backfill without modifying other options in parent object `advanced` (1 ms)
        ✓ should backfill without modifying other options in parent object `advanced.alerts` (1 ms)
        ✓ should backfill without modifying other options in parent object `advanced.alerts.hash` (5 ms)
        ✓ should not backfill if field is already present (1 ms)
      backfilling `events.hash.md5` with `true`
        ✓ should backfill when there are no advanced options yet
        ✓ should backfill without modifying other options in parent object `advanced` (1 ms)
        ✓ should backfill without modifying other options in parent object `advanced.events` (1 ms)
        ✓ should backfill without modifying other options in parent object `advanced.events.hash` (1 ms)
        ✓ should not backfill if field is already present (1 ms)
      backfilling `events.hash.sha1` with `true`
        ✓ should backfill when there are no advanced options yet (1 ms)
        ✓ should backfill without modifying other options in parent object `advanced`
        ✓ should backfill without modifying other options in parent object `advanced.events` (1 ms)
        ✓ should backfill without modifying other options in parent object `advanced.events.hash` (2 ms)
        ✓ should not backfill if field is already present (1 ms)
```